File: Backend/src/main/java/com/example/assignment/service/BlogService.java
package com.example.assignment.service;

import com.example.assignment.entity.Blog;
import com.example.assignment.repository.BlogRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BlogService {

    @Autowired
    private BlogRepository blogRepository;

    public Blog saveDraft(Blog blog) {
        return blogRepository.save(blog);
    }

    public Blog publishBlog(Long id) {
        Blog blog = blogRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Blog not found with ID: " + id));
        blog.setPublished(true);
        return blogRepository.save(blog);
    }

    public List<Blog> getAllBlogs() {
        return blogRepository.findAll();
    }

    public Blog getBlogById(Long id) {
        return blogRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Blog not found with ID: " + id));
    }
}

File: Frontend/src/services/blogService.js
import axios from 'axios';

const API = 'http://localhost:8080/api/blogs';

export const saveDraft = (blog) => axios.post(`${API}/draft`, blog);
export const publishBlog = (id) => axios.post(`${API}/publish/${id}`);
export const getAllBlogs = () => axios.get(API);
export const getBlogById = (id) => axios.get(`${API}/${id}`);
